aptos-move/framework/aptos-framework/sources/stake.move

```diff
@@ -893,11 +893,6 @@ module aptos_framework::stake {
                 continue
             };
 
-            // Restrict a validator's voting power to the max stake allowed.
-            if (new_validator_info.voting_power > maximum_stake) {
-                new_validator_info.voting_power = maximum_stake;
-            };
-
```

Comment:
I don't know why, this is pretty trivial logic.

Reply:
The complexity is if we need to mint to both active and pending_inactive, the code would become a bit more complex to make sure we don't mint rewards to more than max stake amount allowed.

Reply:
When active + pending_inactive > max_stake, we'd need to:
1. Make sure the total rewards minted <= max rewards for max stake.
2. Divide the rewards minted proportionally between active and pending_inactive.